api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -117,11 +117,22 @@ type NodeGroup struct {
 	// MachineConfigPoolSelector defines label selector for the machine config pool
 	// +optional
 	MachineConfigPoolSelector *metav1.LabelSelector `json:"machineConfigPoolSelector,omitempty"`
+	// NodeSelector defines label selector for the nodes that will host RTE pods
+	// +optional
+	NodeSelector *NodeSelector `json:"nodeSelector,omitempty"`
```

Comment:
can this be just a `metav1.LabelSelector`?

Reply:
I wanted a way to name the nodeselector daemonset. For MCP selector nodegroup the ds name is unique for every matching MCP because it ends with the MCP name.
I'm against depending on the labelSelector key-vals because it will turn out very long and unpractical, especially when having more than a single key-value pair. 
I am open for suggestions, we can make it serial (ngns1,ngns2...), name it based on date/time or keep it up to the user to define it as suggested here to choose the name of the nodes group.

Reply:
In addition, we need to add the new fields in `NUMAResourcesOperatorStatus`